test/test_unit.jl

```diff
@@ -2744,6 +2744,101 @@ end
+
+    tspan = (0.0, 1.5) # Re-used for wrapping `rhs` below
+
+    # Call `semidiscretize` to create the ODE problem to have access to the
+    # initial condition based on which the sparsity pattern is computed
+    ode_jac_type = semidiscretize(semi_jac_type, tspan)
+    u0_ode = ode_jac_type.u0
+    du_ode = similar(u0_ode)
+
+    ###############################################################################
+    ### Compute the Jacobian sparsity pattern ###
+
+    # Only the parabolic part of the `SplitODEProblem` is treated implicitly so we only need the parabolic Jacobian, see
+    # https://docs.sciml.ai/DiffEqDocs/stable/types/split_ode_types/#SciMLBase.SplitFunction
+    # Although we do sparsity detection on the entire RHS (since semi_jac_type depends on both equations and 
+    # equations_parabolic), this is equivalent to doing sparsity detection on the diffusion problem alone
+    # (see next section for a demonstration of this)
+
+    # Wrap the `Trixi.rhs_parabolic!` function to match the signature `f!(du, u)`, see
+    # https://adrianhill.de/SparseConnectivityTracer.jl/stable/user/api/#ADTypes.jacobian_sparsity
+    rhs_parabolic_wrapped! = (du_ode, u0_ode) -> Trixi.rhs_parabolic!(du_ode, u0_ode, semi_jac_type, tspan[1])
+
+    jac_prototype_parabolic = jacobian_sparsity(rhs_parabolic_wrapped!, du_ode, u0_ode, jac_detector)
+    ###############################################################################
+    ### Compare sparsity pattern based on hyperbolic-parabolic ###
+    ### problem with sparsity pattern of parabolic-only problem ###
+
+    # We construct a semidiscretization just as we did previously, but this time with advection_velocity=0
+    equations_hyperbolic_no_advection = LinearScalarAdvectionEquation1D(0)
+    equations_parabolic_no_advection = LaplaceDiffusion1D(diffusivity(), equations_hyperbolic_no_advection)
+
+    semi_jac_type_no_advection = SemidiscretizationHyperbolicParabolic(mesh,
+                                                (equations_hyperbolic_no_advection, 
+                                                equations_parabolic_no_advection),
+                                                initial_condition_convergence_test, solver,
+                                                uEltype = jac_eltype)
+    ode_jac_type_no_advection = semidiscretize(semi_jac_type_no_advection, tspan)
+
+    # Do sparsity detection on our semidiscretization with advection turned off
+    rhs_hyper_para! = (du_ode, u0_ode) -> rhs_hyperbolic_parabolic!(du_ode, u0_ode, semi_jac_type_no_advection, tspan[1])
+
+    jac_prototype_parabolic_no_advection = jacobian_sparsity(rhs_parabolic_wrapped!, du_ode, u0_ode, jac_detector)
+
+    # Given that the stencil for parabolic solvers are always larger than those of hyperbolic solvers,
+    # the sparsity detection will never depend on the hyperbolic part of the problem
```

Comment:
I thought about this again, maybe we could also add a test where we have a non-zero advection, but do the sparsity detection on `rhs_hyperbolic_parabolic`. The sparsity pattern should come for the present implementation (i.e., the choice of the parabolic solver) out identical.

Reply:
Makes sense. I added this to this existing test since I didn't want to copy/paste a lot of the set-up code into another test. Let me know if you prefer them to be separate, though.